### PR TITLE
feat(sysgo): extend validity proposer configs + add DatabaseURL support

### PR DESCRIPTION
**Pull request by Farhad-Shabani**

### CODE REVIEW

File: op-devstack/sysgo/l2_proposer_validity.go
Comment (fakedev9999) [inline op-devstack/sysgo/l2_proposer_validity.go]: nit
```suggestion
		pgPass        = "postgres"
```

File: op-devstack/sysgo/deployer_succinct.go [hunk 1]
```diff
@@ -240,22 +242,29 @@ func (o *Orchestrator) deployOpSuccinctL2OutputOracle(
...
+		"L1_BEACON_RPC":    l1CL.beaconHTTPAddr,
+		"L2_RPC":           strings.ReplaceAll(l2EL.UserRPC(), "ws://", "http://"),
+		"L2_NODE_RPC":      strings.ReplaceAll(l2CL.UserRPC(), "ws://", "http://"),
+		"VERIFIER_ADDRESS": l2Net.deployment.sp1MockVerifier.Hex(),
+		"PRIVATE_KEY":      l1PAOKeyStr,
+		"PROPOSER":         proposerAddr.Hex(),
+		"L1_CONFIG_DIR":    l1CfgDir,
+		"L2_CONFIG_DIR":    l2CfgDir,
+		"OP_SUCCINCT_L2_OUTPUT_ORACLE_CONFIG_PATH": l2ooConfigPath,
+		"L2OO_CONFIG_PATH":                         l2ooConfigPath,
```
Comment (fakedev9999) [inline op-devstack/sysgo/deployer_succinct.go]: If there's a reason can you document why both are needed? Or if not, can you consolidate?

File: op-devstack/sysgo/l2_proposer_validity.go [hunk 2]
```diff
@@ -78,6 +101,54 @@ func WithValidityConfigDirsOption(
...
+	})
+}
+
+func WithVPRangeProofEvmGasLimit(n uint64) ValidityProposerOption {
+	return ValidityProposerOption(func(p devtest.P, id stack.L2ProposerID, cfg *ValidityProposerConfig) {
+		cfg.rangeProofEvmGasLimit = &n
+	})
+}
+
+func WithVPMaxConcurrentProofRequests(n uint64) ValidityProposerOption {
```
Comment (fakedev9999) [inline op-devstack/sysgo/l2_proposer_validity.go:122]:
```
if n == 0 {
    p.Require().Fail("maxConcurrentProofRequests must be > 0")
}
```

nit: Should we consider adding validations like ^? And also add a safe guard on the proposer side as well?

**Review by fakedev9999 — APPROVED** [review]
lgtm

File: op-devstack/sysgo/deployer_succinct.go (hunk 1 above)
Comment (Farhad-Shabani) [reply op-devstack/sysgo/deployer_succinct.go]: [a0b2372](https://github.com/succinctlabs/optimism/pull/319/commits/a0b2372cdafe303f04fc3884f34caa5fa102c5a5)

File: op-devstack/sysgo/l2_proposer_validity.go
Comment (Farhad-Shabani) [reply op-devstack/sysgo/l2_proposer_validity.go]: [dc2f0e2](https://github.com/succinctlabs/optimism/pull/319/commits/dc2f0e214fc155af2294019e718fc08e81aedb39)

File: op-devstack/sysgo/l2_proposer_validity.go (hunk 2 above)
Comment (Farhad-Shabani) [reply op-devstack/sysgo/l2_proposer_validity.go:122]: Very good question!

In test suites, I generally prefer not to add validation at such points. Validation in tests usually shows up as assertions instead. That keeps the tests flexible, lets you intentionally feed adversarial or “sad path” inputs later. Like to verify how the proposer behaves on edge cases or invalid data.

On the proposer side, though, stronger validation and type safety absolutely make sense! similar to how we recently implemented range-splitting configs for fault proofs. That said, this still feels like a low-priority improvement to me, not something to optimize immediately.
